src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt

```diff
@@ -222,6 +228,54 @@ class CHIPToolActivity :
         .show()
   }
 
+  private fun onReturnIntent(intent: Intent) {
+    val appLinkData = intent.data
+    // Require URI schema "mt:"
+    if (!appLinkData?.scheme.equals("mt")) return
+    // Require URI host "modelinfo"
+    if (!appLinkData?.host.equals("modelinfo")) return
```

Comment:
Might be more clear to log some errors in these cases

Reply:
Detailed logging added